frontend/src/metabase/dashboard/components/Sidebar.tsx

```diff
@@ -0,0 +1,63 @@
+import { t } from "ttag";
+import type { ReactNode } from "react";
+import ButtonDeprecated from "metabase/core/components/Button";
+import { Button, Icon } from "metabase/ui";
+import {
+  ButtonContainer,
+  ChildrenContainer,
+  SidebarAside,
+} from "./Sidebar.styled";
+
+interface SidebarProps {
+  closeIsDisabled?: boolean;
+  children: ReactNode;
+  onClose?: () => void;
+  onCancel?: () => void;
+  onRemove?: () => void;
+  "data-testid"?: string;
+}
+
+export function Sidebar({
+  closeIsDisabled,
+  children,
+  onClose,
+  onCancel,
+  onRemove,
+  "data-testid": dataTestId,
+}: SidebarProps) {
+  return (
+    <SidebarAside data-testid={dataTestId} $width={384}>
+      <ChildrenContainer>{children}</ChildrenContainer>
+      {(onClose || onCancel || onRemove) && (
+        <ButtonContainer>
+          {onRemove && (
+            <Button
+              leftIcon={<Icon name="trash" />}
+              variant="subtle"
+              color="error"
+              onClick={onRemove}
+              style={{ paddingLeft: 0, paddingRight: 0 }}
```

Comment:
Since we're probably awaiting a transition from mantine 6 to 7, I am adding an inline style, which will be replaced with a CSS class once migrated.